keras/src/backend/tensorflow/layer.py

```diff
@@ -62,17 +62,73 @@ def _trackable_children(self, save_type="checkpoint", **kwargs):
             self.test_function = test_function
             self.predict_function = predict_function
 
-            for tracked_attr in self._tracked:
-                tracked_item = getattr(self, tracked_attr)
-                if isinstance(tracked_item, tracking.TrackedList):
-                    children[tracked_attr] = list(tracked_item)
-                if isinstance(tracked_item, tracking.TrackedDict):
-                    children[tracked_attr] = dict(tracked_item)
-                if isinstance(tracked_item, tracking.TrackedSet):
-                    children[tracked_attr] = list(tracked_item)
+            # Convert Keras tracked collections to plain Python structures
+            # without creating TensorFlow trackable dependencies
+            self._convert_tracked_collections(children)
 
         return children
 
+    @tf.__internal__.tracking.no_automatic_dependency_tracking
```

Comment:
Couldn't you just put `@tf.__internal__.tracking.no_automatic_dependency_tracking` on `def _trackable_children`?